.github/workflows/test.yml

```
@@ -0,0 +1,145 @@
+name: test
+
+on:
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+
+jobs:
+  # Test the Android ds2 binary on an emulator.
+  android-test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        # We don't currently have a good way of running arm64-v8a tests on the
+        # available GitHub test runners.
+        abi: ['x86_64']
+
+    env:
+      ANDROID_API_LEVEL: 28
+      DEBUG_PORT: 5432
+      SCCACHE_DIRECT: yes
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: seanmiddleditch/gha-setup-ninja@v5
+
+      # Free up some disk space by removing some things we don't use. Without
+      # this step, we exhaust disk space while installing Android the emulator
+      # image in the setup-ndk step below. The list of packages sorted by on
+      # disk size can be generated by running the command:
+      #
+      #     dpkg-query -Wf '${Installed-Size}\t${Package}\n' | sort -n
+      #
+      - name: free-up disk space
+        run: |
+          df -h /home
+          sudo apt purge -q aspnetcore-* \
+                            azure-* \
+                            containerd.io \
+                            containernetworking-plugins \
+                            docker-ce* \
+                            dotnet-* \
+                            firefox \
+                            gfortran-* \
+                            google-chrome-stable \
+                            google-cloud-cli* \
+                            kubectl \
+                            linux-azure-* \
+                            linux-modules-*-azure \
+                            microsoft-edge-stable \
+                            mono-* \
+                            monodoc-* \
+                            msbuild \
+                            mysql-* \
+                            podman \
+                            postgresql-* \
+                            powershell \
+                            r-base-core \
+                            vim-runtime
+          sudo apt autoremove
```

```suggestion
          sudo rm -rf /usr/share/swift
          sudo apt autoremove
```

THe Swift toolchain is installed by extracting it to this location, it is not a system package.